.circleci/initjob/40-set-label-environment-variables.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+shared_env=$1
+
+if [[ -f pr-metadata/labels/build-legacy-probes ]]; then
+    echo 'export BUILD_LEGACY_PROBES=true' >> "$shared_env"
+else
+    echo 'export BUILD_LEGACY_PROBES=false' >> "$shared_env"
+fi
+
+if [[ -f pr-metadata/labels/build-full-images ]]; then
+    echo 'export BUILD_FULL_IMAGES=true' >> "$shared_env"
+else
+    echo 'export BUILD_FULL_IMAGES=false' >> "$shared_env"
+fi
+
+if [[ -f pr-metadata/labels/debug ]]; then
+    echo 'export CMAKE_BUILD_TYPE=Debug' >> "$shared_env"
+else
+    echo 'export CMAKE_BUILD_TYPE=Release' >> "$shared_env"
+fi
+
+# Loops over all possible types of Valgrind tests, checks the labels, and sets environment variables accordingly
+# Default is not to use Valgrind, but build with Valgrind if any Valgrind label is present
+BUILD_USE_VALGRIND="false"
+for test_type in unit-tests integration-tests; do
+    CHECK_VALGRIND="false"
+    for valgrind_test_type in helgrind valgrind; do
+        up_valgrind_test_type=${valgrind_test_type^^}
+        up_test_type="${test_type//-/_}"
+        up_test_type="${up_test_type^^}"
+        if [[ -f "pr-metadata/labels/${valgrind_test_type}-${test_type}" ]]; then
```

Comment:
A general question about CI: we have a `pr_has_label.sh` script that seems to do a fair bit of work to find out the labels, and we also have this much simpler approach. Which should we use? If the simple approach, what is the `pr_has_label.sh` script for?

Reply:
It looks like pr_has_label.sh is used in only one place and I don't think it is a good solution as it seems to tie everything that uses it into CircleCI. I have removed it. I think all of these labels should be read into environment variables and those environment variables should be used instead, but that should be done in another PR. I have created a ticket for this https://issues.redhat.com/browse/ROX-9998

Reply:
We already have read-labels.sh which is somewhat redundant with pr_has_label.sh.